wrapper/src/main/java/software/amazon/jdbc/PluginServiceImpl.java

```diff
@@ -365,18 +384,37 @@ protected EnumSet<NodeChangeOptions> compare(
   }
 
   @Override
-  public List<HostSpec> getHosts() {
+  public List<HostSpec> getAllHosts() {
     return this.hosts;
   }
 
+  @Override
+  public List<HostSpec> getHosts() {
+    CustomEndpointInfo customEndpointInfo =
+        this.getInfo(this.initialConnectionHostSpec.getHost(), CustomEndpointInfo.class, true);
+    if (customEndpointInfo == null) {
+      return this.hosts;
+    }
+
+    if (STATIC_LIST.equals(customEndpointInfo.getMemberListType())) {
```

Comment:
I'd prefer to keep cutsom endpoint related classes and types in the CustomEndpoint plugin. PluginServiceImpl should have no references to them.

Reply:
As discussed, this class no longer has references to classes in the customendpoint package